scripts/validate_docstrings.py

```diff
@@ -497,6 +498,20 @@ def parameter_desc(self, param):
                 desc = desc[: desc.index(full_directive)]
         return desc
 
+    def has_proper_punctuation(self, param):
```

Comment:
I'd prefer to call this `param_desc_ends_with_period`.

Also, I'd pass `param_name` and `param_desc` and avoid calling `self.parameter_desc` from this function.